ui/js/component/formField/view.jsx

```diff
@@ -51,6 +56,13 @@ class FormField extends React.PureComponent {
   }
 
   componentDidMount() {
+    if (this.props.regexp) {
+      this._validateCallback = () => this.validate();
+    }
+
+    this.refs.field.addEventListener("blur", this._validateCallback);
+    this.refs.field.form.addEventListener("submit", this._validateCallback);
```

Comment:
How does `this.refs.field.form` always exist?

Reply:
Don't all form inputs have a `.form` property, right? Even `<select>` and such? So as long as `this.refs.field` is some kind of form input, it should work. And because it's in `componentDidMount`, we don't have to worry about the ref not existing.

Reply:
1) I think that is not true. Fields can be rendered outside of forms.
2) Why would being in `componentDidMount` avoid an exception from calling a method on undefined?

Reply:
1. True. I had the sense that having inputs outside of forms was supposed to be a bit of an anti-pattern, since wrapping everything in a `<form>` is more semantic even when you're not submitting to the server. But it looks like there are uses that are considered valid. It sounds like we're going to nix this altogether, but if we somehow end up keeping it, I'll make sure that it doesn't try to interact with the form if the form doesn't exist.
2. It doesn't. I was assuming there'd be a form, being in `componentDidMount` just assures that the ref will already exist.